generator/generator.go

```diff
@@ -522,6 +524,18 @@ func unescapeComment(comment string) string {
 	return val
 }
 
+// getNumberBase will return the base of the number if it is prefixed with 0x, 0, or 0b
+func getNumberBase(value string) (string, int) {
+	if strings.HasPrefix(strings.ToLower(value), "0x") {
+		return value[2:], 16
+	} else if strings.HasPrefix(strings.ToLower(value), "0b") {
+		return value[2:], 2
+	} else if strings.HasPrefix(value, "0") && len(value) > 1 {
+		return value[1:], 8
+	}
```

Comment:
I'm not sure I understand the benefit of this case.  Do you mind elaborating a bit on why you would want this?

Reply:
My intention is to support different representations of values in different binaries, in my project I usually use hexadecimal identifiers for these numbers, and if I convert to decimal for these values, it would be a problem to read the code. Of course, I could add comments to it and state the given value in the comments, but it feels like I'm adding to the problem.

Reply:
0XX is the number written in the base of the eight.